packages/block-editor/src/hooks/use-zoom-out.js

```diff
@@ -9,39 +9,55 @@ import { useEffect, useRef } from '@wordpress/element';
  */
 import { store as blockEditorStore } from '../store';
 import { unlock } from '../lock-unlock';
-
 /**
- * A hook used to set the zoomed out view, invoking the hook sets the mode.
+ * A hook used to set the editor mode to zoomed out mode, invoking the hook sets the mode.
  *
- * @param {boolean} zoomOut If we should zoom out or not.
+ * @param {boolean} zoomOut If we should enter into zoomOut mode or not
  */
 export function useZoomOut( zoomOut = true ) {
-	const { setZoomLevel } = unlock( useDispatch( blockEditorStore ) );
-	const { isZoomOut } = unlock( useSelect( blockEditorStore ) );
+	const { __unstableSetEditorMode, setZoomLevel } = unlock(
+		useDispatch( blockEditorStore )
+	);
+	const { __unstableGetEditorMode } = unlock( useSelect( blockEditorStore ) );
 
-	const originalIsZoomOutRef = useRef( null );
+	const originalEditingModeRef = useRef( null );
+	const mode = __unstableGetEditorMode();
 
 	useEffect( () => {
 		// Only set this on mount so we know what to return to when we unmount.
-		if ( ! originalIsZoomOutRef.current ) {
-			originalIsZoomOutRef.current = isZoomOut();
+		if ( ! originalEditingModeRef.current ) {
+			originalEditingModeRef.current = mode;
 		}
 
-		// The effect opens the zoom-out view if we want it open and the canvas is not currently zoomed-out.
-		if ( zoomOut && isZoomOut() === false ) {
+		return () => {
+			// We need to use  __unstableGetEditorMode() here and not `mode`, as mode may not update on unmount
+			if (
+				__unstableGetEditorMode() === 'zoom-out' &&
+				__unstableGetEditorMode() !== originalEditingModeRef.current
+			) {
+				__unstableSetEditorMode( originalEditingModeRef.current );
+				setZoomLevel( 100 );
+			}
+		};
+	}, [] );
+
+	// The effect opens the zoom-out view if we want it open and it's not currently in zoom-out mode.
+	useEffect( () => {
```

Comment:
I came across this PR while searching for the cause of #66205. I feel like we need to check that it's not a mobile layout here, otherwise the width won't be set correctly when we open the Patterns tab in mobile view.

Reply:
I think we'll need to wait and see if https://github.com/WordPress/gutenberg/pull/66141 gets backported. However, @PARTHVATALIYA's [idea to add it to the inserter useZoomOut conditional](https://github.com/WordPress/gutenberg/issues/66205#issuecomment-2419594065) should work well for either case. If that route will fix the issue, I think that would work well.

Reply:
Confirming that we are _not_ intending to backport https://github.com/WordPress/gutenberg/pull/66141